akka-contrib/src/main/scala/akka/contrib/pattern/ReceivePipeline.scala

```diff
@@ -38,10 +38,13 @@ trait ReceivePipeline extends Actor {
     val around = aroundCache match {
       case Some((`receive`, cached)) ⇒ cached
       case _ ⇒
-        val zipped = pipeline.foldRight[Receive](receive)((outer, inner) ⇒ outer(inner).orElse[Any, Unit](inner))
+        val zipped = pipeline.foldRight[Receive] {
+          case msg => super.aroundReceive(receive, msg)
```

Comment:
I like the idea @jeremystone, seems to be a definitive solution for the flow control problem. I don't think it is a problem to change the API since it is in the contrib module.